### DISCUSSION TITLE
generate DTO from JPA entity ??

### DISCUSSION THREAD

**Discussion by anasoid** [mapstruct/mapstruct#3760] [Q&A] [answered]
Hello,

There is any tool (maven , gradle , ide..) that can generate or Help to generate DTO from source classes like JPA entity, that can be configured by annotation or yaml file..

I knew there is jpa buddy, but this  feature need ultimate license.

Thanks

**Comment by merge-simpson** [comment]
Even if it's not provided via annotations or YAML files, I think this is an easy way.  
**It will work with most of implementations of JPA.** (e.g., hibernates)

<br />

**Use record to simplify your work**:

```java
// JDK 16+ (14 Preview)
public record BoardSummary(String title, BoardStatus status) {
}
```

It would work like:

```java
public interface BoardSummaryProjection {
    String getTitle();
    BoardStatus getStatus();
}
```

Or like this:

```java
@Getter // lombok
public class BoardSummaryProjection {
    private String title;
    private BoardStatus getStatus();
}
```

<br />

<table>
  <tr><td>👍 I'm not a good English speaker. I hope you can understand.</td></tr>
</table>

**Comment by anasoid** [comment]
hello @merge-simpson 

thanks for answer, but your it's not clear for me :D.

I need to generate DTO classes from JPA classes

**Reply by merge-simpson** [reply]
I don't know those kinds of libraries yet.
But can you explain the purpose of that work?

It seems that you need a generator to make DTO classes. Then what is the purpose?

I don't assure that it will be clean codes to make those classes automatically. So I hope to know if there are other reasons. (If I were you, I'd have made DTOs using record because it is as easy as properties or annotations.)

If I can get more idea of yours, it can be a good idea.

**Comment by anasoid** [comment]
hello,

I have  a simple use, i need a DTO class to be used with mapstruct to be converted to JPA class, this DTO classes will be used for import , DTO classes and JPA class are very similar, i home there is a maven or gradle library that i can customize template to generate my DTO.

Idea already have similar tools  https://www.jetbrains.com/help/idea/jpa-buddy-dto-generator.html

**Reply by filiphr** [reply, marked as answer]
@anasoid, I think you are asking in the wrong place. You are asking us to share with you a library that can generate DTOs from your JPA classes during the build time. You'll have to search the internet for this
